samples/excel/85-preview-apis/workbook-calculation.yaml

```diff
@@ -0,0 +1,180 @@
+order: 1
+id: excel-workbook-calculation
+name: Calculations
+description: 'Demonstrates the calculation APIs of the workbook: worksheet-level calulcation events and application-level calculation controls.'
+host: EXCEL
+api_set:
+    ExcelApi: '1.11'
+script:
+    content: |
+        $("#setup").click(() => tryCatch(setup)); $("#register-onCalculated-handler").click(() => tryCatch(registerOnCalculatedHandler)); $("#recalculate-single").click(() => tryCatch(recalculateSingle)); $("#recalculate-column").click(() => tryCatch(recalculateColumn)); $("#manual-calculations").click(() => tryCatch(switchToManualCalculations)); $("#automatic-calculations").click(() => tryCatch(switchToAutomaticCalculations)); $("#force-calculation").click(() => tryCatch(forceCalculation));
+        async function registerOnCalculatedHandler() {
+          await Excel.run(async (context) => {
+            let sheet = context.workbook.worksheets.getItem("Sample");
+            sheet.onCalculated.add(onCalculated);
+            await context.sync();
+            console.log("Added worksheet selection changed event handler.");
+          });
+        }
+        async function onCalculated(event: Excel.WorksheetCalculatedEventArgs) {
+          await Excel.run(async (context) => {
+            console.log(`The Range ${event.address} has recalculated.`);
+          });
+        }
+        async function recalculateSingle() {
+          await Excel.run(async (context) => {
+            const sheet = context.workbook.worksheets.getItem("Sample");
+            const randomRow = Math.floor(Math.random() * 6);
+            const randomColumn = Math.floor(Math.random() * 4);
+            const randomResult = Math.floor(Math.random() * 10000);
+
+            const cellToChange = sheet.getRange("B2:E7").getCell(randomRow, randomColumn);
+
+            cellToChange.load("address");
+            await context.sync();
+            console.log(`Changing cell ${cellToChange.address}`);
+
+            cellToChange.values = [[randomResult]];
+          });
```

Comment:
Have we stopped adding an explicit final `context.sync` to the end of `Excel.run`? We always used to have that as a best practice even though it isn't necessary, because under the hood Excel.run will call context.sync if there are any unsynced commands in the queue.

Reply:
Good call. I added syncs to the end of those write methods. I left it out of methods that looked like this:
```
...
  await context.sync();
  console.log("Something");
});
```

Does that seem reasonable?

Reply:
Yes.